event/main_listener.php

```diff
@@ -224,15 +231,16 @@ public function disable_media_embed($event)
 	 * setup a cache directory to improve scraping performance
 	 *
 	 * @param \phpbb\event\data $event The event object
+	 * @return void
 	 */
 	public function setup_cache_dir($event)
 	{
-		/** @var \phpbb\textformatter\s9e\parser $service  */
-		$service = $event['parser'];
-		$parser = $service->get_parser();
-
-		if ($this->config->offsetGet('media_embed_enable_cache') && $this->cache_dir)
+		if ($this->cache_dir && $this->config->offsetGet('media_embed_enable_cache'))
```

Comment:
I wouldn't probably change the order of checking here as `$this->cache_dir` is set in 99.99% of cases unlike `config['media_embed_enable_cache']`, but ok as far as it works both ways.